PR: https://github.com/odoo/odoo/pull/60235

From: 8b74d39f71519899fa707ff5cadf4bba42394c0e
From: Swapnesh Shah
Date: 2020-10-19 08:42:11

Structural Changes: 6
Total Changes: 6

[FIX] sale_management: correct decimal precision

Before this commit, It was refering to non-existing decimal precision "Product UoS" which doesn't exists since odoo 9.

Followup on d152106b1b8ee8294000bbca7d2ccc9c80a8d05e

closes odoo/odoo#60235

X-original-commit: 9199bd588109330344aa505c5930f127a3920f5e
Signed-off-by: Nicolas Martinelli (nim) <nim@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_management/models/sale_order.py
+++ b/addons/sale_management/models/sale_order.py
@@ -205,7 +205,7 @@ class SaleOrderOption(models.Model):
     discount = fields.Float('Discount (%)', digits='Discount')
     uom_id = fields.Many2one('uom.uom', 'Unit of Measure ', required=True, domain="[('category_id', '=', product_uom_category_id)]")
     product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id', readonly=True)
-    quantity = fields.Float('Quantity', required=True, digits='Product UoS', default=1)
+    quantity = fields.Float('Quantity', required=True, digits='Product Unit of Measure', default=1)
     sequence = fields.Integer('Sequence', help="Gives the sequence order when displaying a list of optional products.")
 
     @api.depends('line_id', 'order_id.order_line', 'product_id')

--- a/addons/sale_management/models/sale_order_template.py
+++ b/addons/sale_management/models/sale_order_template.py
@@ -112,7 +112,7 @@ class SaleOrderTemplateLine(models.Model):
     product_id = fields.Many2one(
         'product.product', 'Product', check_company=True,
         domain=[('sale_ok', '=', True)])
-    product_uom_qty = fields.Float('Quantity', required=True, digits='Product UoS', default=1)
+    product_uom_qty = fields.Float('Quantity', required=True, digits='Product Unit of Measure', default=1)
     product_uom_id = fields.Many2one('uom.uom', 'Unit of Measure', domain="[('category_id', '=', product_uom_category_id)]")
     product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id', readonly=True)
 
@@ -163,7 +163,7 @@ class SaleOrderTemplateOption(models.Model):
         required=True, check_company=True)
     uom_id = fields.Many2one('uom.uom', 'Unit of Measure ', required=True, domain="[('category_id', '=', product_uom_category_id)]")
     product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id', readonly=True)
-    quantity = fields.Float('Quantity', required=True, digits='Product UoS', default=1)
+    quantity = fields.Float('Quantity', required=True, digits='Product Unit of Measure', default=1)
 
     @api.onchange('product_id')
     def _onchange_product_id(self):
